http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/internal/NettyTransportServiceComponent.java

```diff
@@ -62,9 +62,7 @@ protected void activate(BundleContext bundleContext) {
             policy = ReferencePolicy.DYNAMIC,
             unbind = "removeMessageProcessor")
     protected void addMessageProcessor(CarbonMessageProcessor carbonMessageProcessor) {
-        NettyTransportContextHolder.getInstance().getListenerConfigurations().values().forEach(
-                listenerConfiguration -> NettyTransportContextHolder.getInstance().addMessageProcessor(
-                        listenerConfiguration.getPort(), carbonMessageProcessor));
+        NettyTransportContextHolder.getInstance().addMessageProcessor(carbonMessageProcessor);
```

Comment:
This method should be setMessageProcessor because we can have only a single message processor